### ISSUE TITLE
[BUG] websockets dont add or pass custom headers

### ISSUE THREAD

**Issue by si458** [tobychui/zoraxy#426]
**Describe the bug**
websockets dont add or pass custom headers

**To Reproduce**
1. run meshcentral
2. proxy meshcentral
3. inspect http headers for images, they show custom headers that you set or pass on original request
4. inspect websocket headers, they DONT contain custom headers that you set or pass on original request

**Expected behavior**
websocket headers to include your original custom headers AND custom headers you set inside the web ui of zoraxy

**Screenshots**

**Browser (if it is a bug appears on the UI section of the system):**
 - OS: [e.g. iOS] ALL
 - Browser [e.g. chrome, safari] ALL
 - Version [e.g. 22] ALL

**Host Environment (please complete the following information):**
-  Arch: [e.g. arm64] X64
 - Device: [e.g. Bananapi R2 PRO] VM
 - OS: [e.g. Armbian] Ubuntu
 - Version [e.g.  23.02 Bullseye ] 22.04
 - Docker Version (if you are running Zoraxy in docker): [e.g. 3.0.4] 3.1.4

**Additional context**

**Comment by tobychui** [comment]
@si458 Thanks for the investigation. Custom headers are currently **only** applied to HTTP requests (after all, it is a HTTP reverse proxy). I think I can add custom header supports into websocket as well, will take a look at the issue this weekend.

**Comment by si458** [comment]
@tobychui thank you! This will explain why our meshcentralrouter app and meshctrl.js app don't work, because we have to send custom headers on the websocket

Would it be possible to just PASS any existing headers then over the websocket, rather than messing around with adding custom headers?

Only issue we have is our custom header `x-meshauth` contains encoded credentials so it won't be static it will always change

**Comment by tobychui** [comment]
@si458 I need to have a check at the source code though, as I remember current implementation of websocket proxy only copies a limited sets of valid headers (for websocket) from the original request.

https://github.com/tobychui/zoraxy/blob/4d163fe80fe3636b7ea863f8545a65d5906159cd/src/mod/websocketproxy/websocketproxy.go#L123-L136

Though I understand that some projects might want to sneak in some custom headers like in this case `x-meshauth`. I think I will handle that case-by-case (or maybe just add a toggle switch for copying all header from src to dest (unsafe))

**Comment by si458** [comment]
@tobychui a toggle option would be perfectly fine by me! 

One other tiny bug (might be related)
Is the `user-agent` is still showing as `Go-http-client`, and again we pass a custom user-agent to identify application versions and custom applications but again it never sends it :(

**Comment by tobychui** [comment]
@si458 I think this should be fixed in the latest commit, but since I don't have anything to test it on, would you mind testing out the v3.1.5 build from the dev branch (build from source and run it on some kind of test bench) to see if it works?

**Comment by si458** [comment]
@tobychui do you have a dev docker image at all?
otherwise ill look at trying to building it and testing later 👍

**Comment by si458** [comment]
ok the error `RangeError: Invalid WebSocket frame: invalid status code 1006` is actually coming from the websocket disconnects

however i tested nginx and traefik and these dont produce these errors?
its only zoraxy that generate these errors? 
so im not sure how zoraxy handles the disconnects/closes of the websockets but it seems to think they arent standard

**Comment by tobychui** [comment]
@si458 This snippet shows how websocket close is handled in Zoraxy websocket proxy router. 

https://github.com/tobychui/zoraxy/blob/4d163fe80fe3636b7ea863f8545a65d5906159cd/src/mod/websocketproxy/websocketproxy.go#L218-L258

If you are getting 1006 error, you can try enable the Verbal Mode by modifying this section in the top of websocketproxy.go and test it again. You should see the error from Zoraxy's STDOUT. 

```
// Create a new websocket proxy
	wsprox := &WebsocketProxy{Backend: backend, Verbal: true, Options: options} <-- Set verbal to true here
	if options.CopyAllHeaders {
		wsprox.Director = DefaultDirector
	}
```

**Comment by tobychui** [comment]
Hi @si458,

The changes in this issue was merged into v3.1.5 but we are seeing a lot of issues reported by Proxmox and HA users regarding new websocket behavior. So I decided to moved the option to `HTTP Proxy > Edit > Custom Header > Advance Settings > WebSocket Custom Headers` for your specific use case.

![圖片](https://github.com/user-attachments/assets/4e5d1a27-b100-43e1-a747-9e7eb4e106a8)

This change will be included in the next (v3.1.6) release. It would be nice if you could notify your users as well. If you got any issues related to Zoraxy and MeshCentral, please feel free to let me know :)

**Comment by si458** [comment]
Thanks, @tobychui !

One gent already said 3.1.5 works a treat now for him!

However, we are still seeing the same error as explained above https://github.com/tobychui/zoraxy/issues/426#issuecomment-2539048050

I've double-checked our source code, and this error never appears with Apache,nginx,cloudflare,traefik,etc

It only appears with zoraxy

So, meshcentral still thinks the websockets when disconnected don't format correctly

I'm not a GO expert, sadly, so I can not debug your application
